src/genn/genn/runtime/runtime.cc

```diff
@@ -700,6 +700,11 @@ void Runtime::stepTime()
 //----------------------------------------------------------------------------
 void Runtime::customUpdate(const std::string &name)
 {
+    // If NVTX markers are enabled, push range
+    if(m_Backend.get().isNVTXMarkerEnabled()) {
+        m_Backend.get().pushNVTXRange("customUpdate:" + name);
+    }
+
```

Comment:
I think profiling ranges should be inserted in the generated code as otherwise GeNN itself will need to be linked against NVTX. This functions to push and pop ranges will need calling in https://github.com/genn-team/genn/blob/master/src/genn/backends/single_threaded_cpu/backend.cc#L666 and https://github.com/genn-team/genn/blob/master/src/genn/genn/code_generator/backendCUDAHIP.cc#L756